tests/ui/use/use-path-segment-kw.rs

```diff
@@ -0,0 +1,177 @@
+        use $m::$crate as _dollar_crate3; //~ ERROR `$crate` in paths can only be used in start position
+        use $m::{$crate}; //~ ERROR `$crate` in paths can only be used in start position
+        use $m::{$crate as _nested_dollar_crate3}; //~ ERROR `$crate` in paths can only be used in start position
+
+        use crate::$crate; //~ ERROR `$crate` in paths can only be used in start position
+        use crate::$crate as _dollar_crate4; //~ ERROR `$crate` in paths can only be used in start position
+        use crate::{$crate}; //~ ERROR `$crate` in paths can only be used in start position
+        use crate::{$crate as _nested_dollar_crate4}; //~ ERROR `$crate` in paths can only be used in start position
+
+        use super::$crate; //~ ERROR `$crate` in paths can only be used in start position
+        use super::$crate as _dollar_crate5; //~ ERROR `$crate` in paths can only be used in start position
+        use super::{$crate}; //~ ERROR `$crate` in paths can only be used in start position
+        use super::{$crate as _nested_dollar_crate5}; //~ ERROR `$crate` in paths can only be used in start position
+
+        use self::$crate; //~ ERROR `$crate` in paths can only be used in start position
+        use self::$crate as _dollar_crate6; //~ ERROR `$crate` in paths can only be used in start position
+        use self::{$crate}; //~ ERROR `$crate` in paths can only be used in start position
+        use self::{$crate as _nested_dollar_crate6}; //~ ERROR `$crate` in paths can only be used in start position
+
+        use $crate::$crate; //~ ERROR `$crate` in paths can only be used in start position
+        use $crate::$crate as _dollar_crate7; //~ ERROR `$crate` in paths can only be used in start position
+        use $crate::{$crate}; //~ ERROR `$crate` in paths can only be used in start position
+        use $crate::{$crate as _nested_dollar_crate7}; //~ ERROR `$crate` in paths can only be used in start position
+    }
+}
+
+fn outer() {}
+
+mod foo {
+    pub mod bar {
+        pub mod foobar {
+            pub mod qux {
+                pub use super::inner;
+            }
+
+            pub mod baz {
+                pub use super::inner;
+            }
+
+            pub fn inner() {}
+        }
+
+        // --- $crate ---
+        macro_dollar_crate!(foobar);
+
+        // --- crate ---
+        use crate; //~ ERROR crate root imports need to be explicitly named: `use crate as name;`
+        pub use crate as _crate; // Good
+
+        use ::crate; //~ ERROR `crate` in paths can only be used in start position
+        use ::crate as _crate2; //~ ERROR `crate` in paths can only be used in start position
+        use ::{crate}; //~ ERROR `crate` in paths can only be used in start position
+        use ::{crate as _nested_crate2}; //~ ERROR `crate` in paths can only be used in start position
+
+        use foobar::crate; //~ ERROR `crate` in paths can only be used in start position
```

Comment:
So one source of issues in import resolution is that paths are split into two parts in import contexts.
`use foo::bar::baz` -> `foo::bar + baz`, `use foo::bar` -> `foo + bar`, `use foo` -> `<nothing> + foo`.
In other contexts this split doesn't happen, so the behavior may differ.

It would be good to accompany every import-context path in this test with an equivalent non-import context path.
Like
```rust
use foobar::crate;
type A = foobar::crate;
// or
type A = foobar::crate::ExistingType;
// to avoid "expected type, found module" errors, *if* that's more convenient.
```